### PR TITLE
refactor(vpn): move vpn resources to deprecated

### PR DESCRIPTION
**Pull request by niuzhenguo**
**What this PR does / why we need it**:

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```release-note

```

## PR Checklist

* [x] Tests added/passed.
* [ ] Documentation updated.
* [ ] Schema updated.

## Acceptance Steps Performed

```
make testacc TEST='./huaweicloud' TESTARGS='-run=TestAccVpn'
...
=== RUN   TestAccVpnGroupV2_basic
=== PAUSE TestAccVpnGroupV2_basic
=== RUN   TestAccVpnIKEPolicyV2_basic
=== PAUSE TestAccVpnIKEPolicyV2_basic
=== RUN   TestAccVpnIKEPolicyV2_withLifetime
=== PAUSE TestAccVpnIKEPolicyV2_withLifetime
=== RUN   TestAccVpnIPSecPolicyV2_basic
=== PAUSE TestAccVpnIPSecPolicyV2_basic
=== RUN   TestAccVpnIPSecPolicyV2_withLifetime
=== PAUSE TestAccVpnIPSecPolicyV2_withLifetime
=== RUN   TestAccVpnServiceV2_basic
=== PAUSE TestAccVpnServiceV2_basic
=== RUN   TestAccVpnSiteConnectionV2_basic
=== PAUSE TestAccVpnSiteConnectionV2_basic
=== CONT  TestAccVpnGroupV2_basic
=== CONT  TestAccVpnIPSecPolicyV2_withLifetime
=== CONT  TestAccVpnIKEPolicyV2_withLifetime
=== CONT  TestAccVpnIPSecPolicyV2_basic
--- PASS: TestAccVpnIPSecPolicyV2_withLifetime (66.38s)
=== CONT  TestAccVpnIKEPolicyV2_basic
--- PASS: TestAccVpnIKEPolicyV2_withLifetime (66.59s)
=== CONT  TestAccVpnSiteConnectionV2_basic
--- PASS: TestAccVpnIPSecPolicyV2_basic (66.84s)
=== CONT  TestAccVpnServiceV2_basic
--- PASS: TestAccVpnGroupV2_basic (72.45s)
--- PASS: TestAccVpnServiceV2_basic (53.86s)
--- PASS: TestAccVpnIKEPolicyV2_basic (64.97s)
--- PASS: TestAccVpnSiteConnectionV2_basic (70.17s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/deprecated        136.798s
```

### CODE REVIEW

**Review by chengxiangdong — COMMENTED** [review]
1. Missing the unit test files in `huaweicloud/services/acceptance/deprecated`.
2. After move vpn resources to deprecated, the following structs are recommended to be removed from `huaweicloud/value_specs.go`:
   - VpnIPSecPolicyCreateOpts
   - VpnServiceCreateOpts
   - VpnEndpointGroupCreateOpts
   - VpnIKEPolicyCreateOpts
   - VpnIKEPolicyLifetimeCreateOpts
   - VpnSiteConnectionCreateOpts

**Comment by niuzhenguo** [comment]
@chengxiangdong done

File: huaweicloud/services/acceptance/deprecated/resource_huaweicloud_vpnaas_endpoint_group_v2_test.go [hunk 1]
```diff
@@ -1,21 +1,23 @@
...
-	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/utils/fmtp"
 )
 
 func TestAccVpnGroupV2_basic(t *testing.T) {
 	var group endpointgroups.EndpointGroup
 	resource.ParallelTest(t, resource.TestCase{
-		PreCheck:     func() { testAccPreCheck(t) },
-		Providers:    testAccProviders,
+		PreCheck:     func() { acceptance.TestAccPreCheck(t) },
+		Providers:    acceptance.TestAccProviders,
```
Comment (chengxiangdong) [inline huaweicloud/services/acceptance/deprecated/resource_huaweicloud_vpnaas_endpoint_group_v2_test.go]: `Providers` has been deprecated, use `ProviderFactories: acceptance.TestAccProviderFactories,` is better.
There 3 files need to be replaced.

Comment (niuzhenguo) [reply huaweicloud/services/acceptance/deprecated/resource_huaweicloud_vpnaas_endpoint_group_v2_test.go]: done

**Pull request merged**